### DISCUSSION TITLE
We have routes to endpoints which do appear in the open api spec but still work

### DISCUSSION THREAD

**Discussion by JonruAlveus** [octokit/octokit.net#2502] [Q&A]
I’ve noticed that, at least for repos, we have a 2 ways of identifying a Repo:

* Repo/owner/name/…
* Repositories/id/…

It looks like the repositories ones don’t appear much in the open api spec. Are they deprecated now? Should we be supporting the, at all? Are they something to do with Organizations/Enterpise?

**Comment by timrogers** [comment]
Definitely a great question to ask! ❤️

Have you been able to validate that both versions work? 

If so, does the "repositories" version support an owner and a repo name, or just an "ID"?

I have seen these two different paths mentioned inside GitHub, but I'm not sure on the history behind them. It's clear, from the docs, that we prefer "repos", but I can do some digging to figure out the right way forward.

**Reply by JonruAlveus** [reply]
Thanks for the very prompt feedback! Both versions work, we have the, both plumbed into the repositories and tests that execute them, I haven’t tested to see if repositories work with both (or conversely is repos work with both). I probably can’t prove anything until tomorrow though now unfortunately.

**Reply by timrogers** [reply]
Happy to try it myself in `curl` and let you know how I get on. Whenever I have seen `/repositories/...` as opposed to `/repos/...`, it has always referred to "id" rather than "owner"/"name", but I'm not sure on the significance of that.

**Reply by timrogers** [reply]
I looked into this and both versions work, but have different behaviour. 

For `GET /repositories/:id`, you have to use an integer ID for the repo which is available in a few different places in the API.

This endpoint isn't documented, but a lot of people are using it - including through Octokit.rb. I think we should leave it in place, but I'll try to figure out if it should be in our docs/schema.

**Comment by JonruAlveus** [comment]
- On a similar note, I believe the property HasDownloads is available on NewRepository for users/repo CREATE method but not orgs/org/repo CREATE method. It works on the user one. I’m going to set up a free organisation tomorrow to run the tests against so I will post whether it works or not for orgs. Thought I’d raise it as an oddity though 👍

**Reply by timrogers** [reply]
I just tested it and the `has_downloads` attribute does seem to exist and be supported for organization repos - but I'm not sure what it does. I'll report this to the relevant team.

**Reply by timrogers** [reply]
The team at GitHub are looking into this and figure out what the deal is.
